pilot/pkg/serviceregistry/kube/controller/multicluster.go

```diff
@@ -94,7 +95,17 @@ func NewMulticluster(kc kubernetes.Interface, secretNamespace string, opts Optio
 		secretNamespace:       secretNamespace,
 	}
 	mc.initSecretController(kc)
-
+	if features.CentralIstioD {
+		inClusterConfig, err := rest.InClusterConfig()
```

Comment:
I am not sure how this will work. istiod doesn't need to run in cluster

Reply:
this is used to deal with istiod running in an external cluster.  In this case, we need watch `isito-kubeconfig` secret  as it is in the same cluster with istiod.

Reply:
I also doubt this can work.